include/swift/AST/DiagnosticsSema.def

```diff
@@ -4047,6 +4047,8 @@ NOTE(overridden_required_initializer_here,none,
 // Functions
 ERROR(attribute_requires_function_type,none,
       "@%0 attribute only applies to function types", (StringRef))
+ERROR(optional_closures_are_already_escaping,none,
+      "Optional closures are already @escaping", ())
```

Comment:
I'm on the fence about whether this is accurate enough: `@escaping` is a type annotation that only appears in function parameters. Perhaps we should stick to the old note or say *something* like `@escaping attribute is redundant in parameters of optional function type` instead?

Reply:
I agree that it not so accurate, but in my opinion, it's better to follow @xedin suggestion
And here we are now :)